core/runtime/common/runtime_instances_pool.cpp

```diff
@@ -140,13 +140,24 @@ namespace kagome::runtime {
       const GetCode &get_code,
       const RuntimeContext::ContextParams &config) {
     std::unique_lock lock{pools_mtx_};
-    OUTCOME_TRY(getModule(lock, code_hash, get_code, config));
+    OUTCOME_TRY(getPool(lock, code_hash, get_code, config));
     return outcome::success();
   }
 
+  OptRef<const Module> RuntimeInstancesPoolImpl::getModule(
```

Comment:
Reference would escape from mutex
```suggestion
  std::shared_ptr<Module> RuntimeInstancesPoolImpl::getModule(
```